src/explore-education-statistics-admin/src/pages/release/data/components/ReleaseDataUploadsSection.tsx

```diff
@@ -244,36 +245,68 @@ const ReleaseDataUploadsSection = ({
                               >
                                 Edit title
                               </Link>
-                              <Link
-                                className="govuk-!-margin-right-4"
-                                to={generatePath<ReleaseDataFileReplaceRouteParams>(
-                                  releaseDataFileReplaceRoute.path,
-                                  {
-                                    publicationId,
-                                    releaseId,
-                                    fileId: dataFile.id,
-                                  },
-                                )}
-                              >
-                                Replace data
-                              </Link>
+                              {dataFile.isLinkedToApiDataSet ? (
+                                <Modal
+                                  showClose
+                                  title="Cannot replace data"
+                                  triggerButton={
+                                    <ButtonText>Replace data</ButtonText>
+                                  }
+                                >
+                                  <p>
+                                    This data file has an API data set linked to
+                                    it. Please remove the API data set before
+                                    replacing the data.
+                                  </p>
```

Comment:
As discussed on Slack, it'd be good to have a link to the API data set in this message. I've pushed up a change that replaces `isLinkedToApiDataSet` with `publicApiDataSetId`. We should be able to use this link to the relevant API data set page.

In terms of the message, not entirely sure what the best way of presenting this is. Maybe we could add the link to one of the e.g. 'The data file has an [API data set](#test) linked to it. Please remove the API data set before replacing the data.'.

Alternatively, there could be a 'Go to API data set' link. Will leave it up to you!

Reply:
I've gone with adding a  'Go to API data set' link.